components/gitpod-protocol/src/admin-protocol.ts

```diff
@@ -29,6 +29,9 @@ export interface AdminServer {
     adminIsStudent(userId: string): Promise<boolean>;
     adminAddStudentEmailDomain(userId: string, domain: string): Promise<void>;
     adminGrantExtraHours(userId: string, extraHours: number): Promise<void>;
+
+    adminSettingsGetTelemetry(): Promise<boolean>
```

Comment:
Instead of having a method here for every boolean flag, we should implement something akin [`updateLoggedInUser`](https://github.com/gitpod-io/gitpod/blob/0951ced5ebfb2e1a822b7c3eaa4855f7cad7724c/components/server/src/workspace/gitpod-server-impl.ts#L283-L298).

This avoids the necessity to update the API surface and implementation on every change.

Reply:
Note that it's important to have an explicit allowlist for fields to update, otherwise this is an attack vector.

One way to solve this without further duplication would be to change `installation-admin-protocol.ts` to sth like this:
```
/**
 * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
 * Licensed under the GNU Affero General Public License (AGPL).
 * See License-AGPL.txt in the project root for license information.
 */


const InstallationAdminSettingsPrototype = {
    sendTelemetry: true,
};
export type InstallationAdminSettings = typeof InstallationAdminSettingsPrototype;
export namespace InstallationAdminSettings {
    export function fields(): (keyof InstallationAdminSettings)[] {
        return Object.keys(InstallationAdminSettingsPrototype) as (keyof InstallationAdminSettings)[];
    }
}

export interface InstallationAdmin {
    id: string;
    settings: InstallationAdminSettings;
}
export namespace InstallationAdmin {
    export function createDefault(): InstallationAdmin {
        return {
            id: "",
            settings: {
                ...InstallationAdminSettingsPrototype,
            }
        };
    }
}
```
And use the `InstallationAdminSettings.fields()` method to iterate the shape's properties like in `updateLggedInUser` with the explicit list.